crypto/engine/eng_rdrand.c

```diff
@@ -20,28 +20,15 @@
      defined(__x86_64) || defined(__x86_64__) || \
      defined(_M_AMD64) || defined (_M_X64)) && defined(OPENSSL_CPUID_OBJ)
 
-size_t OPENSSL_ia32_rdrand(void);
+size_t OPENSSL_ia32_rdrand_bytes(unsigned char *buf, size_t len);
 
 static int get_random_bytes(unsigned char *buf, int num)
 {
-    size_t rnd;
-
-    while (num >= (int)sizeof(size_t)) {
-        if ((rnd = OPENSSL_ia32_rdrand()) == 0)
-            return 0;
-
-        *((size_t *)buf) = rnd;
-        buf += sizeof(size_t);
-        num -= sizeof(size_t);
-    }
-    if (num) {
-        if ((rnd = OPENSSL_ia32_rdrand()) == 0)
-            return 0;
-
-        memcpy(buf, &rnd, num);
+    if (num < 0 || (unsigned int)num > (size_t)0-1) {
```

Comment:
If right-hand term of `or` expression is supposed to cover case when sizeof(size_t) is less than sizeof(int), then suggested comparison won't solve it. (size_t)0 will be promoted to int, because 1 is int. As result, `(size_t)0-1` would actually be *signed* value -1. So would `(unsigned int)num > -1` cover the case when sizeof(size_t) is less than sizeof(int)? No. Correct solution is to compare to `(size_t)-1`. [*If* one aims to cover the corner case in question that is.]